### PR TITLE
DI plugged

### PR DESCRIPTION
**Pull request by michaem**
Brought models :core:di, :core:network with NetworkModule

### CODE REVIEW

File: application/core/network/library/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 1]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (stepango) [inline application/core/network/library/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: WAT

File: application/common/extensions/util/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+util(
+    testDependencies = deps(
+        test.junit,
+        test.mockito,
```
Comment (stepango) [inline application/common/extensions/util/build.gradle.kts:4]: why 2 mock frameworks?

File: plugin/src/main/java/Deps.kt [hunk 3]
```diff
@@ -48,13 +48,21 @@ object versions {
...
         const val material = "1.2.0"
+        const val dagger = "2.25.3"
     }
 
     object test {
         const val espresso = "3.2.0"
         const val junit = "4.12"
         const val junit_ext = "1.1.1"
         const val hamcrest = "1.3"
+        const val mockito = "2.2.0"
```
Comment (stepango) [inline plugin/src/main/java/Deps.kt:59]: we need only one mocking framework

File: plugin/src/main/java/Deps.kt [hunk 4]
```diff
@@ -401,12 +409,17 @@ object google {
...
         androidx.transition
     )
-}
-
-object javax {
-    val inject = deps(
+    private val inject = deps(
         "javax.inject:javax.inject:1".dep
     )
+    val dagger = deps(
```
Comment (stepango) [inline plugin/src/main/java/Deps.kt:415]: use transitive deps or supply full list :)

File: plugin/src/main/java/Deps.kt [hunk 5]
```diff
@@ -422,6 +435,14 @@ object test {
     val espresso = deps(
         "androidx.test.espresso:espresso-core:${versions.test.espresso}".dep
     )
+
+    val mockito = transitiveDeps(
+        "com.nhaarman.mockitokotlin2:mockito-kotlin:${versions.test.mockito}"
+    )
+
+    val mockk = deps(
```
Comment (stepango) [inline plugin/src/main/java/Deps.kt:443]: it should have at least some transitive deps

File: plugin/src/main/java/library.kt [hunk 6]
```diff
@@ -1,4 +1,29 @@
...
+import com.stepango.forma.module.UtilModule
+import com.stepango.forma.utils.applyDependencies
+import com.stepango.forma.validation.validate
+import com.stepango.forma.validation.validator
 import org.gradle.api.Project
 
-// Same as impl
-fun Project.library(): Unit = TODO()
\ No newline at end of file
+fun Project.library(
```
Comment (stepango) [inline plugin/src/main/java/library.kt:11]: How it's different from util?

File: plugin/src/main/java/Deps.kt (hunk 4 above)
Comment (emartynov) [reply plugin/src/main/java/Deps.kt:415]: That is the great question! Do you have agreement about it? My personal preference to have full list

Comment (stepango) [reply plugin/src/main/java/Deps.kt:415]: my personal preference is to source an entire list of dependencies but currently, it's a time-consuming process which does not contribute much to MVP we are building. So for now I'm ok with transitive deps, one to the items in roadmap is to build some tool to generate the list

**Comment by emartynov** [comment]
Understand

File: application/core/network/library/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker (hunk 1 above)
Comment (michaem) [reply application/core/network/library/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: Seems, it came from marvel-project. Also, that unit-tests has a lot doubtful tools, including Roboelectric...
I think, we don't need all that for our sample

File: application/common/extensions/util/build.gradle.kts (hunk 2 above)
Comment (michaem) [reply application/common/extensions/util/build.gradle.kts:4]: Will be remove that

File: plugin/src/main/java/Deps.kt (hunk 4 above)
Comment (michaem) [reply plugin/src/main/java/Deps.kt:415]: yea, we use transitive deps only for faster delivery MVP functionality now

File: plugin/src/main/java/library.kt (hunk 6 above)
Comment (michaem) [reply plugin/src/main/java/library.kt:11]: Interesting question) In our  first vision `library` should use only for kotlin sources and tests, but now we have same behaviour for `util`.  And differences are only in availability of other project dependencies. `Util` doesn't have it.

**Review by stepango — APPROVED** [review]

**Pull request merged**